### PR TITLE
Setup scripts

### PR DESCRIPTION
**Pull request by dkarnikis**
1. Removed `install.sh` (switching to up.sh)
2. Added usage and parse functions in utils.sh
3. Fixed Docker issues with environment
4. Fixed Github CI to match latest changes to installation

Signed-off-by: Dimitris Karnikis [dkarnikis@gmail.com](mailto:dkarnikis@gmail.com)

### CODE REVIEW

File: scripts/docker/ubuntu/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
-RUN apt-get update -y
-RUN apt-get install -y git 
+RUN apt-get update -y && apt-get install -y git
 ENV PASH_HOST="docker"
+ENV PASH_TOP=/opt/pash
 # download PaSh
-RUN git clone https://github.com/binpash/pash.git /opt/pash --depth 1
-# install PaSh and setup shell paths
-RUN yes | bash /opt/pash/scripts/install.sh -p
+RUN git clone https://github.com/dkarnikis/pash.git /opt/pash --depth 1 -b pash-setup
```
Comment (tammam1998) [inline scripts/docker/ubuntu/Dockerfile:6]: @dkarnikis Should we be pulling from this repo instead of binpash main here?

Comment (dkarnikis) [reply scripts/docker/ubuntu/Dockerfile:6]: oops. you are absolutely correct.

**Review by tammam1998 — APPROVED** [review]
I tested docker locally for ubuntu and debian and seems to be working without problems

File: scripts/distro-deps.sh [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
 
+read_cmd_args $@
+cd $PASH_TOP
+
+LOG_DIR=install_logs
+mkdir -p $LOG_DIR
+
+# if we aren't running in docker, use sudo to install packages
+if [[ "$PASH_HOST" != "docker" ]]; then
+    export SUDO="sudo"
```
Comment (nvasilakis) [inline scripts/distro-deps.sh:21]: Awesome!

File: scripts/setup-pash.sh [hunk 3]
```diff
@@ -147,3 +144,6 @@ do
         fi
     fi
 done
+
+# running simple test that everything installed fine
+$PASH_TOP/pa.sh -c 'PaSh Installation complete!'
```
Comment (nvasilakis) [inline scripts/setup-pash.sh:149]: "Installation" lowercase 'i' :-)

**Review by nvasilakis — APPROVED** [review]
Awesome, much cleaner!

After merging, I'd also run it through `up.sh`, ideally following the (newly updated) documentation (and update appropriately, if needed).

Great work!

**Comment by dkarnikis** [comment]
Merging to main

**Pull request merged**